hedera-node/hedera-schedule-service-impl/src/test/java/com/hedera/node/app/service/schedule/impl/test/handlers/ScheduleSignHandlerTest.java

```diff
@@ -16,162 +16,68 @@
 import com.hedera.node.app.service.schedule.impl.handlers.ScheduleSignHandler;
-import com.hedera.node.app.spi.fixtures.workflows.FakePreHandleContext;
-import com.hedera.node.app.spi.state.ReadableKVStateBase;
+import com.hedera.node.app.spi.fixtures.Assertions;
 import com.hedera.node.app.spi.workflows.PreCheckException;
 import com.hedera.node.app.spi.workflows.PreHandleContext;
+import com.hedera.node.app.workflows.prehandle.PreHandleContextImpl;
+import edu.umd.cs.findbugs.annotations.Nullable;
 import java.util.Collections;
+import org.assertj.core.api.BDDAssertions;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
-import org.mockito.Mock;
 
 class ScheduleSignHandlerTest extends ScheduleHandlerTestBase {
-    private final ScheduleID scheduleID =
-            ScheduleID.newBuilder().scheduleNum(100L).build();
     private ScheduleSignHandler subject;
 
-    @Mock
-    protected JKey adminJKey;
-
-    @Mock
-    protected ScheduleVirtualValue schedule;
-
-    @Mock
-    protected ReadableKVStateBase<Long, ScheduleVirtualValue> schedulesById;
-
-    protected ReadableScheduleStore scheduleStore;
-    private TransactionBody scheduledTxn;
+    private PreHandleContext realPreContext;
+    private TransactionBody scheduled;
 
     @BeforeEach
-    void setUp() {
-        given(states.<Long, ScheduleVirtualValue>get("SCHEDULES_BY_ID")).willReturn(schedulesById);
-        scheduleStore = new ReadableScheduleStoreImpl(states);
-
-        subject = new ScheduleSignHandler(dispatcher);
+    void setUp() throws Exception {
+        subject = new ScheduleSignHandler();
+        setUpBase();
     }
 
     @Test
     void scheduleSignVanillaNoExplicitPayer() throws PreCheckException {
-        final var txn = scheduleSignTransaction();
-        scheduledTxn = givenSetupForScheduleSign();
-
-        final var context = new FakePreHandleContext(accountStore, txn);
-        context.registerStore(ReadableScheduleStore.class, scheduleStore);
-
-        subject.preHandle(context);
-        assertEquals(scheduler, context.payer());
-        assertEquals(schedulerKey, context.payerKey());
-        assertEquals(Collections.EMPTY_SET, context.requiredNonPayerKeys());
+        final TransactionBody testTransaction = scheduleSignTransaction(null);
+        realPreContext = new PreHandleContextImpl(mockStoreFactory, testTransaction, testConfig, mockDispatcher);
 
-        PreHandleContext innerContext = context.innerContext();
-        basicContextAssertions(innerContext, 0);
-        assertEquals(scheduler, innerContext.payer());
-        assertEquals(schedulerKey, innerContext.payerKey());
+        subject.preHandle(realPreContext);
+        BDDAssertions.assertThat(realPreContext.payer()).isEqualTo(scheduler);
+        BDDAssertions.assertThat(realPreContext.payerKey()).isEqualTo(schedulerKey);
+        BDDAssertions.assertThat(realPreContext.optionalNonPayerKeys()).isNotEqualTo(Collections.emptySet());
```

Comment:
Again, shouldn't we be verifying it has the key we expect instead of just checking that it's not empty?

Reply:
We should, and that will be done in the unit test rework and expansion coming soon.